third_party/pytest/pytest_wrapper.py

```diff
@@ -0,0 +1,48 @@
+import os
+import pathlib
+import sys
+
+import coverage
+import pytest
+
+
+def start_coverage_session() -> coverage.Coverage:
+    coverage_dir = pathlib.Path(os.getenv('COVERAGE_DIR'))
+    coverage_file = coverage_dir / '.coverage'
+    coverage_manifest = pathlib.Path(os.getenv('COVERAGE_MANIFEST'))
+    coverage_sources = coverage_manifest.read_text().splitlines()
+    coverage_session = coverage.Coverage(
+        data_file=str(coverage_file),
+        include=coverage_sources,
+    )
+    coverage_session.start()
+    return coverage_session
+
+
+def finalize_coverage_session(coverage_session: coverage.Coverage) -> None:
+    coverage_session.stop()
+    coverage_session.save()
+    output_file = pathlib.Path(os.getenv('COVERAGE_OUTPUT_FILE'))
+    coverage_session.lcov_report(outfile=output_file)
+
+
+def main() -> None:
```

Comment:
This file should be converted to a template, similar to ros2/test.py.tpl. In there, you'd need to add:

```
import {ament_setup}  # Outside main

{ament_setup}.set_up_ament()

test_outputs_dir = os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR')
if test_outputs_dir:
    os.environ['ROS_HOME'] = test_outputs_dir
    os.environ['ROS_LOG_DIR'] = test_outputs_dir
```

Since we're not going to use ros2cli nor ros2test in here (I think), before pytest is launched, you need to handle domain ID as in https://github.com/ros2/ros_testing/blob/humble/ros2test/ros2test/command/test.py#L43-L45. Since all Bazel tests are isolated per default, we can assume domain_id should always be set.

Reply:
Oh, that part I still need to do.